libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -523,51 +552,75 @@ AP_GPS::update_instance(uint8_t instance)
+        // work out which GPS is the primary, and how many sensors we have
+        uint32_t now = AP_HAL::millis();
+        for (uint8_t i=0; i<GPS_MAX_RECEIVERS; i++) {
+            if (state[i].status != NO_GPS) {
+                num_instances = i+1;
             }
+            if (_auto_switch) {
+                if (i == primary_instance) {
+                    continue;
+                }
+                if (state[i].status > state[primary_instance].status) {
+                    // we have a higher status lock, change GPS
+                    primary_instance = i;
+                    _last_instance_swap_ms = now;
+                    continue;
+                }
 
-            bool another_gps_has_1_or_more_sats = (state[i].num_sats >= state[primary_instance].num_sats + 1);
+                bool another_gps_has_1_or_more_sats = (state[i].num_sats >= state[primary_instance].num_sats + 1);
 
-            if (state[i].status == state[primary_instance].status && another_gps_has_1_or_more_sats) {
+                if (state[i].status == state[primary_instance].status && another_gps_has_1_or_more_sats) {
 
-                uint32_t now = AP_HAL::millis();
-                bool another_gps_has_2_or_more_sats = (state[i].num_sats >= state[primary_instance].num_sats + 2);
+                    bool another_gps_has_2_or_more_sats = (state[i].num_sats >= state[primary_instance].num_sats + 2);
 
-                if ( (another_gps_has_1_or_more_sats && (now - _last_instance_swap_ms) >= 20000) ||
-                     (another_gps_has_2_or_more_sats && (now - _last_instance_swap_ms) >= 5000 ) ) {
-                // this GPS has more satellites than the
-                // current primary, switch primary. Once we switch we will
-                // then tend to stick to the new GPS as primary. We don't
-                // want to switch too often as it will look like a
-                // position shift to the controllers.
-                primary_instance = i;
-                _last_instance_swap_ms = now;
+                    if ( (another_gps_has_1_or_more_sats && (now - _last_instance_swap_ms) >= 20000) ||
+                         (another_gps_has_2_or_more_sats && (now - _last_instance_swap_ms) >= 5000 ) ) {
+                    // this GPS has more satellites than the
+                    // current primary, switch primary. Once we switch we will
+                    // then tend to stick to the new GPS as primary. We don't
+                    // want to switch too often as it will look like a
+                    // position shift to the controllers.
+                    primary_instance = i;
+                    _last_instance_swap_ms = now;
+                    }
                 }
+            } else {
+                primary_instance = 0;
             }
-        } else {
-            primary_instance = 0;
+
+            // copy the primary instance to the blended instance
+            state[GPS_MAX_RECEIVERS] = state[primary_instance];
```

Comment:
You aren't blending here so why is the state being copied to the blended state location? The are other code paths that set _output_is_blended to false, but don't ever update the state array. (Same with the _output_location_corrected for that matter).